commcare_connect/form_receiver/processor.py

```diff
@@ -145,6 +152,13 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
         app_build_version=xform.metadata.app_build_version,
         form_json=xform.raw_form,
     )
+    if (
+        counts["daily"] >= opportunity.daily_max_visits_per_user
```

Comment:
These (or at least the last two), should be based on the claim object for that user since they will vary across users (when budget is added and the end date is extended)